app/models/host.rb

```diff
@@ -87,7 +87,7 @@ class Host < ApplicationRecord
                             :inverse_of => :host
   has_many                  :host_aggregate_hosts, :dependent => :destroy
   has_many                  :host_aggregates, :through => :host_aggregate_hosts
-  has_many :host_hardwares, :class_name => 'Hardware', :source => :hardware, :dependent => :nullify
+  has_many :host_hardwares, :class_name => 'Hardware', :dependent => :nullify
   has_many :vm_hardwares,   :class_name => 'Hardware', :through => :vms_and_templates, :source => :hardware
```

Comment:
does `vm_hardwares` need the same removal of `:source => :hardware` ?

Reply:
no. source is used by the `through`